pkg/k8s/annotations.go

```diff
@@ -0,0 +1,58 @@
+package k8s
+
+import (
+	"context"
+
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AnnotateObject adds the specified annotations to the object
+func AnnotateObject(ctx context.Context, c client.Client, obj client.Object, annotations map[string]string) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		_obj := obj.DeepCopyObject().(client.Object)
```

Comment:
```suggestion
		obj := obj.DeepCopyObject().(client.Object)
```

I don't like variable names that start with an underscore. We can just reassign the `obj` variable here

Reply:
I don't want to reassign `obj` because it's a pointer and I don't want it to cause unforeseen problems. I changed the variable name to `o` although I still think `_obj` is better.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1487/commits/50fbe4ea337b258b3238623e1b722f8916498ea8